enhancements/sig-architecture/63-hosted-addon/README.md

```diff
@@ -0,0 +1,262 @@
+- The hosting cluster must be the same cluster where the managed cluster klusterlet(registration-agent & work-agent) 
+runs
+- After the addon is deployed, the deployment mode cannot be switched
+
+## Design Details
+
+### For addon developers
+
+We propose to add a label `addon.open-cluster-management.io/manifest-type` for the addon agent manifests to mark which 
+manifests should be deployed where in Hosted mode:
+- No matter in Default mode or Hosted mode, all manifests will be deployed on the managed cluster
+
+- When the label does not exist or its value is `hosted-mode-managed`: the manifest will be deployed on the managed 
+cluster in Hosted mode
+- When the label's value is `hosted-mode-hosting`: the manifest will be deployed on the hosting cluster in Hosted mode
+- When the label's value is `hosted-mode-none`: the manifest will not be deployed in Hosted mode
+
+
+<table class="tg">
+<thead>
+  <tr>
+    <th class="tg-0pky" rowspan="2"><br>Label Value</th>
+    <th class="tg-c3ow">Default Mode</th>
+    <th class="tg-c3ow" colspan="2">Hosted Mode</th>
+    <th class="tg-c3ow" rowspan="2"><br>Example</th>
+  </tr>
+  <tr>
+    <th class="tg-c3ow">Managed Cluster</th>
+    <th class="tg-c3ow">Managed Cluster</th>
+    <th class="tg-c3ow">Hosting Cluster</th>
+  </tr>
+</thead>
+<tbody>
+  <tr>
+    <td class="tg-0pky"><span style="font-weight:400;font-style:normal">nonexist | hosted-mode-managed</span></td>
+    <td class="tg-c3ow">Y</td>
+    <td class="tg-c3ow">Y</td>
+    <td class="tg-c3ow"></td>
+    <td class="tg-c3ow">CRD owned by the addon</td>
+  </tr>
+  <tr>
+    <td class="tg-0pky"><span style="font-weight:400;font-style:normal">hosted-mode-hosting</span></td>
+    <td class="tg-c3ow">Y</td>
+    <td class="tg-c3ow"></td>
+    <td class="tg-c3ow">Y</td>
+    <td class="tg-c3ow">deployment</td>
+  </tr>
+  <tr>
+    <td class="tg-0pky"><span style="font-weight:400;font-style:normal">hosted-mode-none</span></td>
+    <td class="tg-c3ow">Y</td>
+    <td class="tg-c3ow"></td>
+    <td class="tg-c3ow"></td>
+    <td class="tg-c3ow"></td>
+  </tr>
+</tbody>
+</table>
+
+### For addon deployers
+
+Once the managed cluster is imported to the hub in `Hosted` mode, we propose to add an annotation 
+`addon.open-cluster-management.io/hosting-cluster-name` for `ManagedClusterAddon` to 
+indicate the addon should be deployed in Hosted mode, and the value of the annotation is the hosting cluster which 
+should:
+- be a managed cluster of the hub as well
+- be the same cluster where the managed cluster klusterlet(registration-agent & work-agent) runs
+
+#### How does the addon agent connect to the managed cluster?
+
+Addon agents deployments will be deployed in the `{ManagedClusterAddon.Spec.InstallNamespace}` namespeace, and the 
+addon deployer is required to provide a secret, which name is `{addon-name}-managed-kubeconfig` and contains the key 
```

Comment:
define what is addon deployer.

Reply:
@qiujian16 should we change the `{addon-name}-managed-kubeconfig` to `{addon-name}-spoke-kubeconfig`?